### ISSUE TITLE
what's the mean of "full_area" and "box-area"

### ISSUE THREAD

**Issue by hust-kevin** [zjhuang22/maskscoring_rcnn#6]
https://github.com/zjhuang22/maskscoring_rcnn/blob/0ae2a7c01171d106ace7063128c717f8fe19faf9/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py#L73
@zjhuang22

**Comment by zjhuang22** [comment]
> [maskscoring_rcnn/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py](https://github.com/zjhuang22/maskscoring_rcnn/blob/0ae2a7c01171d106ace7063128c717f8fe19faf9/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py#L73)
> 
> Line 73 in [0ae2a7c](/zjhuang22/maskscoring_rcnn/commit/0ae2a7c01171d106ace7063128c717f8fe19faf9)
> 
>  full_area = torch.tensor(mask_util.area(rle_for_fullarea).sum().astype(float)) 
> 
> @zjhuang22

Thanks for your attention. mask_targets is only a mask in a bounding box, and a bounding box only occupies part of the ground truth mask, while our mask IOU is the IOU of the predicted mask and ground truth mask rather than the IOU of the mask_targets. So we need to calculate the proportion of mask_targets in the bounding box to the ground truth mask. This bbox_area refers to the area of mask_targes in box, full_area refers to the area of all ground truth masks, and mask_ratio is the ratio of the two. Then the IOU of the predicted mask and ground truth masks is calculated by this ratio.

**Comment by JoyHuYY1412** [comment]
> > [maskscoring_rcnn/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py](https://github.com/zjhuang22/maskscoring_rcnn/blob/0ae2a7c01171d106ace7063128c717f8fe19faf9/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py#L73)
> > Line 73 in [0ae2a7c](/zjhuang22/maskscoring_rcnn/commit/0ae2a7c01171d106ace7063128c717f8fe19faf9)
> > full_area = torch.tensor(mask_util.area(rle_for_fullarea).sum().astype(float))
> > @zjhuang22
> 
> Thanks for your attention. mask_targets is only a mask in a bounding box, and a bounding box only occupies part of the ground truth mask, while our mask IOU is the IOU of the predicted mask and ground truth mask rather than the IOU of the mask_targets. So we need to calculate the proportion of mask_targets in the bounding box to the ground truth mask. This bbox_area refers to the area of mask_targes in box, full_area refers to the area of all ground truth masks, and mask_ratio is the ratio of the two. Then the IOU of the predicted mask and ground truth masks is calculated by this ratio.

But I think you need to calculate the overlap area to represent IOU?

**Comment by zjhuang22** [comment]
> > > [maskscoring_rcnn/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py](https://github.com/zjhuang22/maskscoring_rcnn/blob/0ae2a7c01171d106ace7063128c717f8fe19faf9/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py#L73)
> > > Line 73 in [0ae2a7c](/zjhuang22/maskscoring_rcnn/commit/0ae2a7c01171d106ace7063128c717f8fe19faf9)
> > > full_area = torch.tensor(mask_util.area(rle_for_fullarea).sum().astype(float))
> > > @zjhuang22
> > 
> > 
> > Thanks for your attention. mask_targets is only a mask in a bounding box, and a bounding box only occupies part of the ground truth mask, while our mask IOU is the IOU of the predicted mask and ground truth mask rather than the IOU of the mask_targets. So we need to calculate the proportion of mask_targets in the bounding box to the ground truth mask. This bbox_area refers to the area of mask_targes in box, full_area refers to the area of all ground truth masks, and mask_ratio is the ratio of the two. Then the IOU of the predicted mask and ground truth masks is calculated by this ratio.
> 
> But I think you need to calculate the overlap area to represent IOU?

Yes, in our code the IoU is calculated by overlap/union.
